### PR TITLE
Adding new options for trimgalore

### PR DESCRIPTION
**Pull request by chuan-wang**

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -118,6 +123,16 @@ if (params.rlocation){
...
 
+// Custom trimming options
+params.length = 18
+params.clip_R1 = 0
+params.three_prime_clip_R1 = 0
+
+// Define regular variables so that they can be overwritten
+length = params.length
+clip_R1 = params.clip_R1
+three_prime_clip_R1 = params.three_prime_clip_R1
```
Comment (ewels) [inline main.nf]: No need to do this here if you don't have presets for different library prep kits.

File: main.nf [hunk 2]
```diff
@@ -252,8 +265,11 @@ process trim_galore {
     file "*_fastqc.{zip,html}" into trimgalore_fastqc_reports
 
     script:
+    tg_length = "--length ${params.length}"
+    c_r1 = ${params.clip_R1} > 0 ? "--clip_R1 ${params.clip_R1}" : ''
+    tpc_r1 = ${params.three_prime_clip_R1} > 0 ? "--three_prime_clip_R1 ${params.three_prime_clip_R1}" : ''
```
Comment (ewels) [inline main.nf]: `${}` is only required when in double quotes. So this should work:

```groovy
c_r1 = params.clip_R1 > 0 ? "--clip_R1 ${params.clip_R1}" : ''
tpc_r1 = params.three_prime_clip_R1 > 0 ? "--three_prime_clip_R1 ${params.three_prime_clip_R1}" : ''
```

Comment (chuan-wang) [reply main.nf]: Thanks Phil! /chuan

**Comment by ewels** [comment]
Great! 👍

**Pull request merged**